app/models/nic/base.rb

```diff
@@ -178,6 +178,12 @@ def ip6=(addr)
       super(Net::Validations.normalize_ip6(addr))
     end
 
+    def matches_subnet?(ip_field, subnet_field)
+      return unless send(subnet_field).present?
+      value = send(ip_field)
```

Comment:
`public_send` preferred too, since those are public methods - it reduces the chances of accidentally calling `matches_subnet` with private methods